src/DynamoCoreWpf/Views/Preview/PreviewControl.xaml.cs

```diff
@@ -264,6 +271,52 @@ private void RefreshCondensedDisplay()
             smallContentView.Text = cachedSmallContent; // Update displayed text.
         }
 
+        private void RefreshExpandedDisplayAsync()
+        {
+            var scheduler = nodeViewModel.DynamoViewModel.Model.Scheduler;
+
+            // The preview control will not have its content refreshed unless 
+            // the content is null. In order to perform real refresh, new data 
+            // source needs to be rebound to the preview control by calling 
+            // BindToDataSource method.
+            // 
+            if (this.cachedLargeContent != null)
+                return;
+
+            if (largeContentGrid.Children.Count == 0)
+            {
+                var tree = new WatchTree();
+                tree.DataContext = new WatchViewModel(nodeViewModel.DynamoViewModel.VisualizationManager);
+                largeContentGrid.Children.Add(tree);
+            }
+
+            var watchTree = largeContentGrid.Children[0] as WatchTree;
+            var rootDataContext = watchTree.DataContext as WatchViewModel;
+
+            var task0 = new DelegateBasedAsyncTask(scheduler, () =>
+            {
+                cachedLargeContent = nodeViewModel.DynamoViewModel.WatchHandler.GenerateWatchViewModelForData(
+                    mirrorData, null, string.Empty, false);
+            });
+
+            scheduler.ScheduleForExecution(task0);
```

Comment:
We don't need the second task below as we can count on the completion notification from the first task (do note that the callback for `Completed` event is done on scheduler thread, which is why we still need a `BeginInvoke` call):

```csharp
task0.Completed += delegate(AsyncTask task)
{
    Dispatcher.BeginInvoke((Action) (() =>
    {
        rootDataContext.Children.Add(cachedLargeContent);

        watchTree.treeView1.SetBinding(ItemsControl.ItemsSourceProperty,
            new Binding("Children")
            {
                DynamoModel.MakeConnectionCommand.Mode = BindingMode.TwoWay,
                Source = rootDataContext
            });

        BeginViewSizeTransition(ComputeLargeContentSize());
    }));
};

scheduler.ScheduleForExecution(task0);
```

Please take note that I have moved `BeginViewSizeTransition` from the caller.